pkg/tfpfbridge/provider_resources.go

```diff
@@ -27,15 +27,19 @@ import (
 	"github.com/pulumi/pulumi-terraform-bridge/pkg/tfpfbridge/pfutils"
 )
 
-func (p *Provider) resources(ctx context.Context) (res pfutils.Resources, err error) {
+type resourceHandle struct {
+	makeResource          func() pfresource.Resource
+	terraformResourceName string
+	schema                tfsdk.Schema
+	pulumiResourceInfo    *info.ResourceInfo // optional
+	idExtractor           idExtractor
+}
 
+func (p *Provider) resources(ctx context.Context) (res pfutils.Resources, err error) {
 	p.resourcesOnce.Do(func() {
```

Comment:
Not the topic of this PR but what is `resourcesOnce`?

Reply:
it's making sure resource scanning happens only once lazily on first ask. Should probably move to a constructor instead somewhere.